Specification.md

```diff
@@ -1277,6 +1286,84 @@ Sigsum integration fails, return an HTTP 500 response to the client and abort.
 
 Finally, make the appropriate changes to the local database (based on what action is to be performed).
 
+### TOTP
+
+Every Fediverse instance may enroll a secret key for use with [TOTP (RFC 6238)](https://www.rfc-editor.org/rfc/rfc6238.html).
+If one is enrolled for a given instance, an TOTP challenge will be required in order to successfully issue a 
+[BurnDown](#burndown) protocol message, or to rotate or disenroll the TOTP secret.
+
+Every instance will have one TOTP secret that may be shared across administrators, for the following reasons:
+
+1. The Public Key Directory doesn't know which Actor on a given instance is an administrator, and which isn't.
+2. If an attacker gets root access to your instance, if TOTPs were per-Actor, they could just issue protocol messages
+   from an Actor that has not enrolled a TOTP secret before. (Or enroll a fresh TOTP secret for a desired Actor.)
+
+Instead of per-Actor, we constrain the TOTP secret to affect the entire instance (should anyone choose to issue a
+BurnDown on behalf of other users).
+
+There are four TOTP operations, but none of them will result in a Protocol Message or change to the underlying Merkle
+Tree state.
+
+#### TOTP Configuration
+
+* Time window size: **`30` seconds**
+* Hash function: **`SHA-512`**
+* OTP length: **`8` digits**
+
+Some authenticator apps do not support these parameters.
+
+#### TOTP Enrollment
+
+Clients **MUST** generate at least 192 bits of randomness (preferably 256). This will serve as the TOTP secret key.
```

Comment:
Why not make it 256? 
Also why the client, would it not better to rely on the PKD to generate the secret?
Since then only one implementation has to do it right and the PKD has not to check the length.